pennylane/qnodes/base.py

```diff
@@ -134,19 +141,40 @@ class BaseQNode:
     (corresponding to a :ref:`variational circuit <varcirc>`)
     and the computational device it is executed on.
 
+    The quantum function can take two kinds of classical input parameters:
+    *positional* and *auxiliary*.
+    Parameters that have default values in the quantum function signature are interpreted as
+    auxiliary parameters. All other parameters are positional.
+
+    * The quantum function can *only* be differentiated with respect to its positional parameters.
+      The positional parameters should be only used as the parameters of the
+      :class:`quantum operations <.Operator>` in the function,
+      and they must be real numbers, or nested ``Iterables`` of real numbers
+      (note that this includes NumPy arrays).
+      Classical processing of positional parameters, either by arithmetic operations
+      or external functions, is not allowed. One current exception is simple scalar multiplication.
+      Positional parameters *must* be given using the positional syntax.
+
+    * The auxiliary parameters can *not* be differentiated with respect to.
```

Comment:
This one reads slightly weird, it might be worth wording it similarly as it was previously for `The quantum function can *only* be differentiated with respect to its positional parameters.`